tests/test_workers/test_tasks/test_opm_operations.py

```diff
@@ -155,3 +156,214 @@ def test_serve_cmd_at_port_delayed_initialize(
+    if index_db_file:
+        with open(df_path, 'r') as f:
+            assert any(
+                line.find('/var/lib/iib/_hidden/do.not.edit.db') != -1 for line in f.readlines()
+            )
+    else:
+        with open(df_path, 'r') as f:
+            assert all(
+                line.find('/var/lib/iib/_hidden/do.not.edit.db') == -1 for line in f.readlines()
+            )
+
+
+@pytest.mark.parametrize("set_index_db_file", (False, True))
+@mock.patch('iib.workers.tasks.utils.run_cmd')
+def test_omp_generate_dockerfile_no_dockerfile(mock_run_cmd, tmpdir, set_index_db_file):
+    index_db_file = os.path.join(tmpdir, 'database/index.db') if set_index_db_file else None
+    fbc_dir = os.path.join(tmpdir, 'catalogs')
+    df_path = os.path.join(tmpdir, f"{os.path.basename(fbc_dir)}.Dockerfile")
+
+    with pytest.raises(IIBError, match=f"Can not find generated Dockerfile at {df_path}"):
+        opm_operations.omp_generate_dockerfile(fbc_dir, tmpdir, index_db_file)
+
+    mock_run_cmd.assert_called_once_with(
+        ['opm', 'alpha', 'generate', 'dockerfile', fbc_dir],
+        {'cwd': tmpdir},
+        exc_msg='Failed to generate Dockerfile for file-based catalog',
+    )
+
+
+@pytest.mark.parametrize('from_index', (None, 'some_index:latest'))
+@pytest.mark.parametrize('bundles', (['bundle:1.2', 'bundle:1.3'], []))
+@pytest.mark.parametrize('overwrite_csv', (True, False))
+@pytest.mark.parametrize('container_tool', (None, 'podwoman'))
+@mock.patch('iib.workers.tasks.utils.set_registry_token')
+@mock.patch('iib.workers.tasks.utils.run_cmd')
+def test_opm_registry_add(
+    mock_run_cmd, mock_srt, from_index, bundles, overwrite_csv, container_tool
+):
+    opm_operations._opm_registry_add(
+        base_dir='/tmp/somedir',
+        index_db='/tmp/somedir/some.db',
+        bundles=bundles,
+        from_index=from_index,
+        overwrite_from_index_token='user:pass',
+        overwrite_csv=overwrite_csv,
+        container_tool=container_tool,
+    )
+
+    mock_run_cmd.assert_called_once()
+    opm_args = mock_run_cmd.call_args[0][0]
+    assert opm_args[:3] == ['opm', 'registry', 'add']
+    if bundles:
+        assert ','.join(bundles) in opm_args
+    else:
+        assert '""' in opm_args
+    if overwrite_csv:
+        assert '--overwrite-latest' in opm_args
+    else:
+        assert '--overwrite-latest' not in opm_args
+    if container_tool:
+        assert '--container-tool' in opm_args
+        assert container_tool in opm_args
+    else:
+        assert '--container-tool' not in opm_args
+    assert "--enable-alpha" in opm_args
+
+    mock_srt.assert_called_once_with('user:pass', from_index)
+
+
+@pytest.mark.parametrize('from_index', (None, 'some_index:latest'))
+@pytest.mark.parametrize('bundles', (['bundle:1.2', 'bundle:1.3'], []))
+@pytest.mark.parametrize('overwrite_csv', (True, False))
+@pytest.mark.parametrize('container_tool', (None, 'podwoman'))
+@mock.patch('iib.workers.tasks.opm_operations.omp_generate_dockerfile')
```

Comment:
opm

Reply:
rebase should fix it - this is not related to this PR. Resolving conversation